src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -96,16 +108,44 @@
     /** EntityResolver provider, used in SLD parsing */
     EntityResolverProvider entityResolverProvider;
 
+    /** HTTP client used to fetch remote styles * */
+    CloseableHttpClient httpClient;
     /**
      * This flags allows the kvp reader to go beyond the SLD library mode specification and match
      * the first style that can be applied to a given layer. This is for backwards compatibility
      */
     private boolean laxStyleMatchAllowed = true;
 
     public GetMapKvpRequestReader(WMS wms) {
+        this(wms, null);
+    }
+
+    public GetMapKvpRequestReader(WMS wms, HttpClientConnectionManager manager) {
         super(GetMapRequest.class);
         this.wms = wms;
         this.entityResolverProvider = new EntityResolverProvider(wms.getGeoServer());
+
+        // configure the http client used to fetch remote styles
+        RequestConfig requestConfig = RequestConfig.custom().build();
```

Comment:
Does it make any difference between this an using RequestConfig.DEFAULT directly?